src/protocol/decoder.js

```diff
@@ -56,10 +56,34 @@ module.exports = class Decoder {
     return this.canReadBytes(INT64_SIZE)
   }
 
+  // Implementation taken from Node 12
+  // https://github.com/nodejs/node/blob/923d8bc733262cf960b62a02f113cfb0412b5834/lib/internal/buffer.js#L140
+  readBigInt64BE(buffer, offset = 0) {
```

Comment:
Since this doesn't advance `this.offset`, we shouldn't expose this as a public method of the decoder. I get that essentially you're just "polyfilling" Node's `readBigInt64BE`, but that doesn't belong on the decoder.

Instead, either move it to a util or inline it in `readInt64` and have it read the initial offset from `this.offset` and then update `this.offset` before returning the value.